cgi-bin/LJ/Sysban.pm

```diff
@@ -198,6 +198,42 @@ sub sysban_check {
             }, undef, $wh, $vl);
     };
 
+    # helper for iteratively checking all subdomains of a hostname
+    my $check_all_subdomains = sub {
```

Comment:
It took me most of an hour to wrap my brain around what this was doing, but I think it is essentially correct - it takes every active, unexpired domain ban in the database and looks for either an exact match to the given host, or a wildcard match for a larger domain of which the given host is a subdomain.  I can use this, but I should probably add a comment or two... :blush: